conf/modules.config

```diff
@@ -17,171 +17,294 @@ process {
+                params.flye_args
+            ].join(" ").trim()
+        }
         publishDir = [
-            path: { "$params.outdir/genome_properties/ploidyplot" },
+            path: { "${params.outdir}/assemble/flye/${meta.id}" },
             mode: params.publish_dir_mode,
             saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
         ]
     }
-    withName: 'MERQURYFK_KATGC' {
+    // HIFIASM:
+    /* Don't think I should switch to nf-core 
+       here since the UL process takes more inputs */
+    withName: 'HIFIASM.*' {
+        ext.args = { [ params.hifiasm_args ] }
         publishDir = [
-            path: { "$params.outdir/genome_properties/katgc" },
+            path: { "${params.outdir}/assemble/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}/${meta.id}" },
             mode: params.publish_dir_mode,
             saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
         ]
     }
+    /* 
+    ----------
+    QC
+    ----------
+    */
+    // This:
+    // Subworkflow: "${task.process.tokenize(':')[-3].toLowerCase()}"
+    // Add sub-sub workflow (as /subsubworkflow) if subworkflow is not assemble:
+    // ${task.process.tokenize(':')[2].toLowerCase() == 'assemble' ? '' : '/' + task.process.tokenize(':')[-3].toLowerCase() }
 
-    // COMPARE_LIBRARIES
-    // withName: 'MERQURYFK_KATCOMP' {
-    //     publishDir = [
-    //         path: { "$params.outdir/library_comparison/kat_comp" },
-    //         mode: params.publish_dir_mode,
-    //         saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
-    //     ]
-    // }
+    // BUSCO: nf-core
+    withName: BUSCO {
+        ext.prefix = { "${meta.id}-${task.process.tokenize(':')[-3].toLowerCase()}-${lineage}" }
+        publishDir = [
+            path: { "${params.outdir}/QC/${task.process.tokenize(':')[2].toLowerCase()}${task.process.tokenize(':')[2].toLowerCase() == 'assemble' ? '' : '/' + task.process.tokenize(':')[-3].tokenize('_')[1].toLowerCase()}/busco/${meta.id}" },
+            mode: params.publish_dir_mode,
+            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+        ]
+    }
 
-    // CONTAMINATION SCREEN
-    withName: 'MASH_SCREEN' {
-        ext.args   = '-w'
-        ext.prefix = { "${meta.id}_${query.baseName}" }
+    // QUAST: Prefer to keep the local module since it can deal with the inputs I have
+    withName: QUAST {
+        ext.prefix = { "${meta.id}_${task.process.tokenize(':')[-3].toLowerCase()}" }
         publishDir = [
-            path: { "$params.outdir/contamination_screen/mash_screen/screens" },
+            path: { "${params.outdir}/QC/${task.process.tokenize(':')[2].toLowerCase()}${task.process.tokenize(':')[2].toLowerCase() == 'assemble' ? '' : '/' + task.process.tokenize(':')[-3].tokenize('_')[1].toLowerCase()}/quast/${meta.id}" },
```

Comment:
Is there a necessity for this complex expression? Why not use `meta.id` all the time?

Reply:
QUAST (and the other QC tools), will run after each of the steps in this assembly; i.e. `assembly`, `polish` and `scaffold`, and polish and scaffold may also be different tools. If I only used `meta.id` this would put every output in the same, which may overwrite things, and is a huge mess. This way they go into e.g. `polish/pilon/$meta.id` or `polish/medaka/$meta.id` for the two polishing options.

Reply:
I think it's clearer to use subworkflows then and use a pattern match against the subworkflow to specify this.
```nextflow
process {
    withName: '.*:POLISH:QUAST' {
        publishDir = ...
    }
    withName: '.*:PILON:QUAST' {
        publishDir = ...
    }
```
It also makes it easier to add a tool option to a given stage of using quast.

Reply:
That is much clearer, I agree.